web/src/composables/useTransformTupleData.ts

```diff
@@ -0,0 +1,91 @@
+import { Ref, computed } from "vue"
+import { ModelType } from "@/domain/types"
+
+type FieldData = { [fieldName: string]: ModelType }
+
+function getTupleKey(parentKey: string, childKey: string) {
+  const sentenceCaseChildKey =
+    childKey.charAt(0).toUpperCase() + childKey.slice(1)
+  return `${parentKey}And${sentenceCaseChildKey}`
+}
+
+function getKeysFromTupleKey(combinedTupleKey: string) {
+  const parentKey = combinedTupleKey.substring(
+    0,
+    combinedTupleKey.indexOf("And")
+  )
+  let childKey = combinedTupleKey.substring(
+    combinedTupleKey.indexOf("And") + 3,
+    combinedTupleKey.length
+  )
+  childKey = childKey.charAt(0).toLowerCase() + childKey.slice(1)
+  return { parentKey, childKey }
+}
+
+//TODO Naming to be done
+function transformToNestedData(
+  data: Record<string, ModelType>,
+  parentKey: string,
+  childKey: string
+): Record<string, ModelType> {
+  const nestedData = { ...data }
+  delete nestedData[parentKey]
+  delete nestedData[childKey]
+  const tupleKey = getTupleKey(parentKey, childKey)
+  Object.assign(nestedData, {
+    [tupleKey]: {
+      parent: data[parentKey],
+      child: data[childKey],
+    },
+  })
+  return nestedData
+}
+
+function transformToFlatData(newValues: Record<string, ModelType>) {
+  for (const tupleKey in newValues) {
+    const tupleValue = newValues[tupleKey]
+    if (
+      typeof tupleValue === "object" &&
+      "parent" in tupleValue &&
+      "child" in tupleValue
+    ) {
+      const { parentKey, childKey } = getKeysFromTupleKey(tupleKey)
+
+      const flattenData = { ...newValues }
```

Comment:
I find verbs for constants mostly unsuitable

Reply:
[bab5b93](https://github.com/digitalservicebund/ris-backend-service/pull/295/commits/bab5b93392dccfc1067eeff74d6920b1a7db69d5)